project-guidelines/quality-output/security/README.md

```diff
@@ -1,12 +1,83 @@
 # Security
 
-The security of the project always used to be considered backend’s job. But there is almost the same amount of vulnerabilities present on frontend. Especially with Single Page Applications, where you often deal with sensitive data.
+Security is an important aspect of all modern web applications. Implementing good security practices help prevent and mitigate various security threats and keep the user's data safe. The checklist below includes the common recommendations and sanity checks that will greatly improve security of your web applications.
 
-ESLint has got your back for many security issues. But the best way to protect yourself is to know what kind of attacks could happen. [Hacksplaining.com](https://www.hacksplaining.com) is a great resource to learn how to perform attacks and, more importantly, how to prevent them.
+## General rules
 
-If you are building a server-side rendered app, this also means that you need to take care of server security. The bare minimum is [HelmetJS](https://helmetjs.github.io) for Express servers.
+- Use encrypted protocols everywhere (HTTPS for web traffic, WSS for secure socket connection)
+- Don’t build custom authentication system yourself, use tried and tested libraries instead
+- Keep your dependencies up-to-date to prevent zero-day vulnerabilities
+- Exclude admin and private URLs from indexing and crawling (using robots.txt)
+- Don’t use iterable URLs (e.g. prefer `/users/[guid]` or `/users/[handle]` instead of `/users/[incremented-integer]`)
+- Remove unused functionality instead of hiding it (with `display: none`, for instance)
+- Use DDOS mitigation via a global caching proxy service like CloudFlare
+- For apps containing sensitive data, consider implementing auto session termination after a period of inactivity or require addition password check for destructive actions
+- Set `Cache-Control: no-store, max-age=0` for non-public/non-static content to make sure that the sensitive user-specific data doesn’t remain cached on disk after a logout
+- Add [subresource integrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) checks if loading your JavaScript libraries from a third party CDN
 
-Use amazing [https://observatory.mozilla.org/](https://observatory.mozilla.org/)
+## Server secrets
```

Comment:
@koss-lebedev How about mentioning [@strv/shh](https://github.com/strvcom/shh)?

Reply:
@koss-lebedev: something like this?

> Alternatively to not versioning `.env` files, you can encrypt & version them using [git-crypt](https://github.com/AGWA/git-crypt) – but that would assume anyone with access to the repository should ultimately have access to all secrets, which isn't always the case.